wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -477,55 +477,16 @@ public TypeSpec generateMessage(MessageType type) {
 
     builder.addMethod(messageFieldsConstructor(nameAllocator, type));
     builder.addMethod(messageFieldsAndUnknownFieldsConstructor(nameAllocator, type));
+    // If there are no fields, then we already created a public no-args constructor.
+    if (!type.fieldsAndOneOfFields().isEmpty()) {
+      builder.addMethod(messageNoArgsConstructor(type));
+    }
     builder.addMethod(newBuilder(nameAllocator, type));
     builder.addMethod(messageEquals(nameAllocator, type));
     builder.addMethod(messageHashCode(nameAllocator, type));
 
     if (emitAndroid) {
-      String adapterName = nameAllocator.get("ADAPTER");
```

Comment:
I didn't change this code.  I just moved it because checkstyle complained that this method was more than 150 lines.